backend/app/assets/javascripts/spree/backend/images/upload.js

```diff
@@ -0,0 +1,204 @@
+// Inspired by: http://html5demos.com/dnd-upload
+Spree.prepareImageUploader = function () {
+  var uploadZone = document.getElementById('upload-zone');
+  if(!uploadZone) return;
+
+  var UploadZone = Backbone.View.extend({
+    el: uploadZone,
+
+    events: {
+      "dragover" : "onDragOver",
+      "dragleave" : "onDragLeave",
+      "drop" : "onDrop",
+      'change input[type="file"]' : "onFileBrowserSelect"
+    },
+
+    progressZone: document.getElementById('progress-zone'),
+
+    // Hide or highlight supported browser features
+    initialize: function() {
+      "filereader formdata progress".split(' ').forEach(function (api) {
+        this.support[api].className = (this.tests[api] === false) ? 'red' : 'hidden'
+      }, this);
+    },
+
+    upload: function(file) {
+      if (!this.tests.formdata) return;
+
+      var progressModel = new ProgressModel({file: file});
+      progressModel.previewFile();
+      progressModel.uploadFile();
+
+      var progressView = new ProgressView({model: progressModel});
+      this.progressZone.appendChild(progressView.render().el);
+    },
+
+    onDragOver: function() {
+      this.el.className = 'hover';
+      return false;
+    },
+
+    onDragLeave: function() {
+      this.el.className = '';
+      return false;
+    },
+
+    onDrop: function(e) {
+      this.el.className = '';
+      e.preventDefault();
+
+      for (var i = 0; i < e.originalEvent.dataTransfer.files.length; i++) {
+        this.upload(e.originalEvent.dataTransfer.files[i]);
+      }
+    },
+
+    onFileBrowserSelect: function(e) {
+      for (var i = 0; i < e.target.files.length; i++) {
+        this.upload(e.target.files[i]);
+      }
+    },
+
+    tests: {
```

Comment:
I'm not in favour of adding more libs. The code for those random pieces of text is ~20 links. As you see fit..